src/tools/idlc/src/descriptor.c

```diff
@@ -1287,8 +1289,12 @@ emit_array(
        we're processing a struct type, and this can be used to determine if its
        an external member */
     idl_node_t *parent = idl_parent(node);
-    if (idl_is_struct(stype->node) && idl_is_external(parent))
-      opcode |= DDS_OP_FLAG_EXT;
+    if (idl_is_struct(stype->node)) {
+      if (idl_is_external(parent))
```

Comment:
I'd not mark a string as external. Maybe I am overlooking something, but it only seems like it complicates things.